src/NServiceBus.Core/Transports/Learning/DelayedMessagePoller.cs

```diff
@@ -3,56 +3,60 @@
     using System;
     using System.Globalization;
     using System.IO;
-    using System.Threading;
+    using System.Threading.Tasks;
     using Logging;
 
     class DelayedMessagePoller
     {
         public DelayedMessagePoller(string basePath)
         {
             this.basePath = basePath;
-            delayedMessagePoller = new Timer(MoveDelayedMessagesToMainDirectory);
+            delayedMessagePoller = new AsyncTimer();
```

Comment:
nitpick, but I found this variable name to be confusing as you end up with multiple calls to `delayedMessagePoller.Stop()`  which are actually different objects.